sycl/source/detail/device_image_impl.hpp

```diff
@@ -95,10 +95,14 @@ class device_image_impl {
 
     // Otherwise, if the device candidate is a sub-device it is also valid if
     // its parent is valid.
-    if (!getSyclObjImpl(DeviceCand)->isRootDevice())
-      return has_kernel(KernelIDCand,
-                        DeviceCand.get_info<info::device::parent_device>());
-
+    if (!getSyclObjImpl(DeviceCand)->isRootDevice()) {
+      try {
+        return has_kernel(KernelIDCand,
+                          DeviceCand.get_info<info::device::parent_device>());
+      } catch (std::exception &e) {
+        __SYCL_REPORT_EXCEPTION_TO_STREAM("exception in has_kernel", e);
```

Comment:
This will be a user-visible message, so it should have more information

Reply:
The exception details comes from the origin (far down the call chain).
This function is to catch it and exit gracefully with whatever propagated from the exception origin.
There is no way at this outermost level can specify the details of the exception.